lib/scenic/adapters/postgres/views.rb

```diff
@@ -0,0 +1,56 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Fetches defined views from the postgres connection.
+      # @api private
+      class Views
+        def initialize(connection)
+          @connection = connection
+        end
+
+        # All of the views that this connection has defined.
+        #
+        # This will include materialized views if those are supported by the
+        # connection.
+        #
+        # @return [Array<Scenic::View>]
+        def all
+          views + materialized_views
```

Comment:
> Can a nonmaterialized view depend on a materialized view?
Yes.

> Is this going to get us into trouble like we saw previously with views depending on views that were alphabetically earlier than their dependancies?

Someday. This combination here produces no difference from the ordering that was previously in place which was a SQL union of the two queries.

But I take your point and hadn't thought of that. I pretty much figured there was no way we had all this dependency stuff sorted just yet, but I hadn't thought of exactly how we got it wrong. This is one way, for sure.

There're probably a way we can get views and matviews in a single query in the order postgres wants to return them to us (which is, hopefully, the order we'd dump them in, but that's probably not guaranteed either). But I dont think that's a change for here.